File: python/create_user.py
# Sample python program to call Tazapay sandbox API
​
import hashlib
import base64
import json
import requests
from datetime import datetime
import calendar
from random import *
import hmac
​
access_key = 'your-access-key'
secret_key = 'your-secret-key'
salt = 'Tif6r2ZqVn' # any salt input
​
http_method = 'POST' # upper case
base_url = 'https://api-sandbox.tazapay.com'
path = '/v1/user' # This can hold any value of the supported API.
​
# Current Unix time.
d = datetime.utcnow()
timestamp = calendar.timegm(d.utctimetuple())
​
# generate signature
to_sign = http_method + path + salt + str(timestamp) + access_key + secret_key
h = hmac.new(bytes(secret_key, 'utf-8'), bytes(to_sign, 'utf-8'), hashlib.sha256)
signature = base64.urlsafe_b64encode(str.encode(h.hexdigest()))
​
headers = {
    'accesskey': access_key,
    'signature': signature,
    'salt': salt,
    'timestamp': str(timestamp)
}
​
payload = {
    "email": "support@tazapay.com",
    "country": "Singapore",
    "ind_bus_type": "Business",
    "business_name":"business name"
}
​
response = requests.post(url=base_url+path, json=payload, headers=headers)
print(json.loads(response.content.decode('utf-8')))

File: go/create_user.go
package main

import (
	"crypto/hmac"
	"crypto/rand"
	"crypto/sha256"
	"encoding/base64"
	"encoding/hex"
	"fmt"
	"io/ioutil"
	"net/http"
	"strings"
	"time"
)

//genrateRandomString
func genrateRandomString(len int) (string, error) {
	b := make([]byte, len)
	if _, err := rand.Read(b); err != nil {
		return "", err
	}
	s := fmt.Sprintf("%X", b)
	return s, nil
}

// generateTimeStamp
func generateTimeStamp() int64 {
	return time.Now().Unix()
}

func generateSignatureString(path, method, signtureSalt, accessKey, secretKey, timestamp string) string {
	return fmt.Sprintf("%s%s%s%s%s%s", strings.ToUpper(method), path, signtureSalt, timestamp, accessKey, secretKey)
}

func generateSHA56(data string, secret string) string {
	h := hmac.New(sha256.New, []byte(secret))
	h.Write([]byte(data))
	sha := hex.EncodeToString(h.Sum(nil))
	return sha
}

func GenerateSignature(path, method, salt, accessKey, secret, timestamp string) string {
	signatureString := generateSignatureString(path, method, salt, accessKey, secret, timestamp)
	signature := generateSHA56(signatureString, secret)
	signature = base64.StdEncoding.EncodeToString([]byte(signature))
	return signature
}

func main() {
	salt, err := genrateRandomString(12)
	if err != nil {
		fmt.Println("err")
		return
	}
	timeStamp := generateTimeStamp()
	accessKey := "your access key"
	secretKey := "your secret key"
	singnature := GenerateSignature("/v1/user", "POST", salt, accessKey, secretKey, fmt.Sprintf("%d", timeStamp))
	url := "https://api-sandbox.tazapay.com"
	method := "POST"

	// country input format is ISO 3166 alpha-2
	// optional fields: contact_code, contact_number, partners_customer_id
	payload := strings.NewReader(`{
		"email": "abc@gmail.com",
		"country": "IN",
		"ind_bus_type": "Business",
		"business_name": "Waynes industry",
		"contact_code": "",
		"contact_number": "",
		"partners_customer_id": ""
	}
`)

	client := &http.Client{}
	req, err := http.NewRequest(method, url, payload)

	if err != nil {
		fmt.Println(err)
		return
	}
	req.Header.Add("accesskey", accessKey)
	req.Header.Add("signature", singnature)
	req.Header.Add("salt", salt)
	req.Header.Add("timestamp", fmt.Sprintf("%d", timeStamp))
	req.Header.Add("Content-Type", "application/json")

	res, err := client.Do(req)
	if err != nil {
		fmt.Println(err)
		return
	}
	defer res.Body.Close()

	body, err := ioutil.ReadAll(res.Body)
	if err != nil {
		fmt.Println(err)
		return
	}
	fmt.Println(string(body))
}
